nibabel/brainvoyager/bv.py

```diff
@@ -0,0 +1,957 @@
+
+    This recursive function returns the dot product of all spatial
+    transformation matrices given in st_array for applying them in one go.
+    The order of multiplication follow the order in the given array.
+
+    Parameters
+    ----------
+    st_array: array of shape (n, 4, 4)
+        array filled with n transformation matrices of shape (4, 4)
+
+    inv: boolean
+        Set to true to invert the transformation matrices before
+        multiplication.
+
+    Returns
+    -------
+    combined_st : array of shape (4, 4)
+    """
+    if len(st_array) == 1:
+        if inv:
+            return np.linalg.inv(st_array[0])
+        else:
+            return st_array[0]
+    if inv:
+        return np.dot(np.linalg.inv(st_array[0, :, :]),
+                      combine_st(st_array[1:, :, :], inv=inv))
+    else:
+        return np.dot(st_array[0, :, :],
+                      combine_st(st_array[1:, :, :], inv=inv))
+
+
+def parse_st(st_dict):
+    """Parse spatial transformation stored in a BV header OrderedDict.
+
+    This function parses a given OrderedDict from a BV header field and returns
+    a spatial transformation matrix as a numpy array.
+
+    Parameters
+    ----------
+    st_dict: OrderedDict
+        OrderedDict filled with transformation matrices of shape (4, 4)
+
+    Returns
+    -------
+    st_array : array of shape (4, 4)
+    """
+    if st_dict['nr_of_trans_val'] != 16:
+        raise BvError('spatial transformation has to be of shape (4, 4)')
+    st_array = []
+    for v in range(st_dict['nr_of_trans_val']):
+        st_array.append(st_dict['trans_val'][v]['value'])
+    return np.array(st_array).reshape((4, 4))
+
+
+class BvError(Exception):
+    """Exception for BV format related problems.
+
+    To be raised whenever there is a problem with a BV fileformat.
+    """
+
+    pass
+
+
+class BvFileHeader(Header):
+    """Class to hold information from a BV file header."""
+
+    # Copies of module-level definitions
+    _data_type_codes = data_type_codes
+    _field_recoders = {'datatype': data_type_codes}
+
+    # format defaults
+    # BV files are radiological (left-is-right) by default
+    # (VTC files have a flag for that, however)
+    default_x_flip = True
+    default_endianness = '<'  # BV files are always little-endian
+    allowed_dtypes = [1, 2, 3]
+    default_dtype = 2
+    allowed_dimensions = [3]
+    data_layout = 'C'
+    hdr_dict_proto = BV_HDR_DICT_PROTO
+
+    def __init__(self,
+                 hdr_dict=None,
+                 endianness=default_endianness,
+                 check=True,
+                 offset=None):
+        """Initialize header from binary data block.
+
+        Parameters
+        ----------
+        binaryblock : {None, string} optional
+            binary block to set into header.  By default, None, in
+            which case we insert the default empty header block
+        endianness : {None, '<','>', other endian code} string, optional
```

Comment:
I exchanged the binaryblock part of the docstring with a hdr_dict description. This is needed for setting parameters in the init call and also allows for easy duplication of BV files. Or maybe I do not get your intention here... :)